crates/bevy_render/src/renderer/mod.rs

```diff
@@ -128,17 +129,38 @@ const GPU_NOT_FOUND_ERROR_MESSAGE: &str = if cfg!(target_os = "linux") {
     "Unable to find a GPU! Make sure you have installed required drivers!"
 };
 
+const REQUESTED_GPU_NOT_FOUND_ERROR_MESSAGE: &str =
+    "WGPU_ADAPTER_NAME set but no matching adapter found!";
+
 /// Initializes the renderer by retrieving and preparing the GPU instance, device and queue
 /// for the specified backend.
 pub async fn initialize_renderer(
     instance: &Instance,
     options: &WgpuSettings,
     request_adapter_options: &RequestAdapterOptions<'_, '_>,
 ) -> (RenderDevice, RenderQueue, RenderAdapterInfo, RenderAdapter) {
-    let adapter = instance
-        .request_adapter(request_adapter_options)
-        .await
-        .expect(GPU_NOT_FOUND_ERROR_MESSAGE);
+    let adapter = {
+        if let Ok(desired_adapter_name) = std::env::var("WGPU_ADAPTER_NAME")
+            .as_deref()
+            .map(str::to_lowercase)
+        {
+            let adapters = instance.enumerate_adapters(Backends::all());
```

Comment:
Doesn't this ignore the passed `request_adapter_options` if `WGPU_ADAPTER_NAME` is set? So `compatible_surface` is not checked to be compatible with the requested adapter.